docs/developer-docs/latest/update-migration-guides/migration-guides/v4/migration-guide-4.0.x-to4.0.6.md

```diff
@@ -6,7 +6,7 @@ canonicalUrl: https://docs.strapi.io/developer-docs/latest/update-migration-guid
 
 # v4.0.x to v4.0.6 migration guide
 
-The Strapi v4.0.x to v4.0.6 migration guide upgrades all prior versions of v4.0.x to v4.0.6. The migration adds the `session` middleware to the middleware array and configures the `session` middleware. The upgrade is required for the [Users & Permissions providers](/user-docs/latest/settings/configuring-users-permissions-plugin-settings.md) to function properly, secure cookies, and encrypt data. The migration guide consists of 3 sections:
+The Strapi v4.0.x to v4.0.6 migration guide upgrades all prior versions of v4.0.x to v4.0.6. The migration adds the `session` middleware to the middleware array and configures the `session` middleware. The `session` middleware is based on [koa-session](https://github.com/koajs/session/blob/master/Readme.md) and is necessary to fix the login provider feature of the Users & Permissions plugin. Additionally, password protection in the Documentation plugin uses the `session` middleware. The migration guide consists of 3 sections:
```

Comment:
I'd add more links to our own documentation:
- the session middleware documentation (see https://docs.strapi.io/developer-docs/latest/setup-deployment-guides/configurations/required/middlewares.html#session) 
- and appropriate links to the Users & Permissions plugin and Documentation plugin documentation

Then, it won't be necessary anymore to link to koa session here (it's explained in the session middleware docs) :-)